docs/source/FEATURES/causal_reasoning.md

```diff
@@ -0,0 +1,78 @@
+# 🧠 Causal Reasoning
```

Comment:
Can you please update this section to reflect the latest changes (see [here](https://github.com/ExtensityAI/symbolicai/issues/71#issuecomment-2349682936)).
Please update the main README + the docs to reflect the following idea:

> The first symbol (in the above case that's s1) must have the only_nesy flag set to True for the logical operators. Without going into detail for why we did this, the tl'dr is that we didn't find a better way in python to do meta-overloading. Without this flag, the logical operators default to string concatenation. For us, having syntactic sugar for concatenation is important, such as being able to use operators like 'A' & 'B' & 'C' to produce 'ABC' or 'A' | 'B' | 'C' to result in 'A B C '.

Feel free to ChatGPT it to make the text clearer.